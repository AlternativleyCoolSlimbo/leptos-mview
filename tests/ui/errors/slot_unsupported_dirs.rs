use leptos::*;
use leptos_mview::mview;

#[slot]
struct Nothing {}

#[component]
fn TakesNothing(nothing: Nothing) -> impl IntoView { let _ = nothing; }

fn try_bad_dirs() {
    let attrs: Vec<(&'static str, Attribute)> = Vec::new();
    let _spread = mview! {
        TakesNothing {
            slot:Nothing {..attrs};
        }
    };

    let _on = mview! {
        TakesNothing {
            slot:Nothing on:click={|_| ()};
        }
    };

    let _attr = mview! {
        TakesNothing {
            slot:Nothing attr:something="something";
        }
    };

    fn a_directive(_el: HtmlElement<html::AnyElement>) {}
    let _use = mview! {
        TakesNothing {
            slot:Nothing use:a_directive;
        }
    };

    let _prop = mview! {
        TakesNothing {
            slot:Nothing prop:value="1";
        }
    };
}

fn main() {}
