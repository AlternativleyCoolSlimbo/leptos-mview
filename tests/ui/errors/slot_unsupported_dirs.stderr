error: spread attributes not supported on components/slots
  --> tests/ui/errors/slot_unsupported_dirs.rs:14:26
   |
14 |             slot:Nothing {..attrs};
   |                          ^^^^^^^^^

error: `on:` is not supported on slots
  --> tests/ui/errors/slot_unsupported_dirs.rs:20:26
   |
20 |             slot:Nothing on:click={|_| ()};
   |                          ^^

error: `attr:` is not supported on slots
  --> tests/ui/errors/slot_unsupported_dirs.rs:26:26
   |
26 |             slot:Nothing attr:something="something";
   |                          ^^^^

error: `use:` is not supported on slots
  --> tests/ui/errors/slot_unsupported_dirs.rs:33:26
   |
33 |             slot:Nothing use:a_directive;
   |                          ^^^

error: `prop:` is not supported on components/slots
  --> tests/ui/errors/slot_unsupported_dirs.rs:39:26
   |
39 |             slot:Nothing prop:value="1";
   |                          ^^^^
